server/region_syncer/client.go

```diff
@@ -186,18 +187,25 @@ func (s *RegionSyncer) StartSyncWithLeader(addr string) {
 				}
 				stats := resp.GetRegionStats()
 				regions := resp.GetRegions()
+				regionLeaders := resp.GetRegionLeaders()
 				hasStats := len(stats) == len(regions)
 				for i, r := range regions {
-					var region *core.RegionInfo
+					var (
+						region       *core.RegionInfo
+						regionLeader *metapb.Peer
+					)
+					if regionLeaders[i].Id != 0 {
```

Comment:
I guess you need to make it compatible with older version.
When rolling update, new version and old version can exist at the same time. Since old PD will not send any leader, the `regionLeaders` can be zero-length. So, it seems check length instead can make it compatible.

Reply:
The `SyncRegionResponse.RegionLeaders` should have the same length as `SyncRegionResponse.Regions` to maintain correspondence. But indeed I should check whether `regionLeaders` is nil or have zero length.